src/jdk.attach/windows/classes/sun/tools/attach/AttachProviderImpl.java

```diff
@@ -42,10 +44,11 @@ public AttachProviderImpl() {
             throw new RuntimeException(
                 "This provider is not supported on this version of Windows");
         }
-        String arch = System.getProperty("os.arch");
-        if (!arch.equals("x86") && !arch.equals("amd64") && !arch.equals("aarch64")) {
+        if (!Architecture.isX86() &&
+            !Architecture.isX64() &&
+            !Architecture.isAARCH64()) {
             throw new RuntimeException(
-                "This provider is not supported on this processor architecture");
+                "This provider is not supported on this processor architecture: " + Architecture.current());
```

Comment:
Could this constructor be changed to be a no-op? The check for Windows 9 and Windows Me dates back to JDK 6 and is no longer needed. The os.arch check also dates from JDK 6 when the  attach provider didn't work on IA64 The only Windows builds now are windows-x86, windows-x64, windows-aarch64 so I assume this check can go away.

Reply:
Removed.

Reply:
> Removed.

Good, that makes it simpler and it means the qualified export can be removed too.